pkg/operator/k8sutil/node.go

```diff
@@ -65,14 +65,18 @@ func ValidNode(node v1.Node, placement rookalpha.Placement) (bool, error) {
 	return ValidNodeNoSched(node, placement)
 }
 
-// GetValidNodes returns all nodes that (1) are not cordoned, (2) meet Rook's placement terms, and
-// (3) are ready.
-func GetValidNodes(rookStorage rookalpha.StorageScopeSpec, clientset kubernetes.Interface, placement rookalpha.Placement) []rookalpha.Node {
+// GetValidNodes returns desiredNodes, validDesiredNodes
+// desiredNodes are rookStorage.Nodes updated to better reflect the corresponding k8s objects:
+// validDesiredNodes are a subset of desiredNodes that are:
+//     - are not cordoned
+//     - meet Rook's placement terms
+//     - are Ready.
+func GetValidNodes(rookStorage rookalpha.StorageScopeSpec, clientset kubernetes.Interface, placement rookalpha.Placement) ([]rookalpha.Node, []rookalpha.Node) {
```

Comment:
I would like to see the changes to this file reverted entirely. This pretty clearly seems to be a change intended only for Ceph, but by changing this file, it is changing behavior for any backend. And, this is breaking the rules of single-purpose functions. 

It seems that the StorageScopeSpec may be somewhat broken if it needs to be updated to better reflect the k8s objects. Maybe a better way would be to write a separate function similar to the ones used to fully resolve nodes.